pkg/validation/validate_cluster.go

```diff
@@ -234,23 +235,23 @@ func (v *ValidationCluster) validateNodes(cloudGroups map[string]*cloudinstances
 				continue
 			}
 
-			role := util.GetNodeRole(node)
-			if role == "" {
-				role = "node"
+			roles := util.GetNodeRoles(node)
+			// Default to node
+			if roles == "" {
+				roles = "node"
 			}
 
 			n := &ValidationNode{
 				Name:     node.Name,
 				Zone:     node.ObjectMeta.Labels["failure-domain.beta.kubernetes.io/zone"],
 				Hostname: node.ObjectMeta.Labels["kubernetes.io/hostname"],
-				Role:     role,
+				Roles:    roles,
 				Status:   getNodeReadyStatus(node),
 			}
 
 			ready := isNodeReady(node)
 
-			// TODO: Use instance group role instead...
-			if n.Role == "master" {
+			if strings.Contains(n.Roles, "master") {
```

Comment:
Using a set or slice also means we don't break if e.g. some creates a role called "notmaster" :-)